mlflow/models/evaluation/base.py

```diff
@@ -643,6 +645,98 @@ def _get_last_failed_evaluator():
+    }
+
+    for metric_name in validation_thresholds.keys():
+
+        metric_threshold, validation_result = (
+            validation_thresholds[metric_name],
+            validation_results[metric_name],
+        )
+
+        if metric_name not in candidate_metrics:
+            validation_result.missing = True
+            continue
+
+        candidate_metric_value, baseline_metric_value = (
+            candidate_metrics[metric_name],
+            baseline_metrics[metric_name] if baseline_metrics else None,
+        )
+
+        # If metric is higher is better, >= is used, otherwise <= is used
+        # for thresholding metric value and model comparsion
+        comparator_fn = operator.__ge__ if metric_threshold.higher_is_better else operator.__le__
+
+        if metric_threshold.threshold is not None:
+            # metric threshold failed
+            # - if not (metric_value >= threshold) for higher is better
+            # - if not (metric_value <= threshold) for lower is better
+            validation_result.threshold_failed = not comparator_fn(
+                candidate_metric_value, metric_threshold.threshold
+            )
+
+        if not baseline_metrics or metric_name not in baseline_metrics:
+            continue
+
+        if metric_threshold.min_absolute_change is not None:
+            # metric comparsion aboslute change failed
+            # - if not (metric_value >= baseline + min_absolute_change) for higher is better
+            # - if not (metric_value <= baseline + min_absolute_change) for lower is better
+            validation_result.min_absolute_change_failed = not comparator_fn(
+                Decimal(candidate_metric_value),
+                Decimal(baseline_metric_value + metric_threshold.min_absolute_change),
+            )
+
+        if metric_threshold.min_relative_change is not None:
+            # metric comparsion relative change failed
+            # - if (metric_value - baseline) / baseline < min_relative_change for higher is better
+            # - if (baseline - metric_value) / baseline < min_relative_change for lower is better
+            if metric_threshold.higher_is_better:
+                relative_change = (
+                    candidate_metric_value - baseline_metric_value
+                ) / baseline_metric_value
+            else:
+                relative_change = (
+                    baseline_metric_value - candidate_metric_value
+                ) / baseline_metric_value
+            validation_result.min_relative_change_failed = (
+                relative_change < metric_threshold.min_relative_change
+            )
+
+    failure_messages = []
+
+    for metric_validation_result in validation_results.values():
+        if metric_validation_result.is_success():
+            continue
+        failure_messages.append(str(metric_validation_result))
+
+    if not failure_messages:
+        return
+
+    raise MlflowException(message=os.linesep.join(failure_messages))
```

Comment:
Nit: Can we add a 4xx error code that seems reasonable here? I think it would also be helpful to use a custom exception that inherits from `MlflowException` and is specific to the model validation use case, e.g. `ModelValidationFailedException`.

Reply:
sounds good, what error code should we use, `400 Bad Request`?